### DISCUSSION TITLE
useAnimatedKeyboard layout ERROR when device is iOS and has SafeAreaView

### DISCUSSION THREAD

**Discussion by LuisTejedaDev** [software-mansion/react-native-reanimated#6154] [General]
When a device like the iPhone 15 Pro opens its keyboard and I make use of useAnimatedKeyboard to move my elements so that the keyboard does not cover them as for example a TextInput, this one presents a failure in the design since the space that occupies the inferior SafeAreaView is not considered by the useAnimatedKeyboard, reason why a completely empty space is left between my element and the keyboard.

On Android and devices that do not have this "SafeAreaView" feature there is no problem, but on new devices specifically on iOS they present it.

**Comment by LuisTejedaDev** [comment]
<img width="795" alt="image" src="https://github.com/software-mansion/react-native-reanimated/assets/145240805/c8644953-d775-455a-8783-acdc184f44ab">

**Comment by MatiPl01** [comment]
Hi! Could you please provide a code snippet? It is hard to help you without seeing how you implemented the logic.
